reana_workflow_controller/rest/utils.py

```diff
@@ -434,8 +434,16 @@ def list_directory_files(
     workspace_path: str, search: Dict[str, List[str]] = None
 ) -> List[dict]:
     """Return a list of files inside a given workspace."""
+    if int(os.getenv("REANA_GC_ENABLED", "0")) > 0:
+        gc.collect()
     file_list = []
+    MAX_FILES_DISPLAY = int(os.getenv("REANA_MAX_WORKSPACE_FILES", "100000"))
```

Comment:
DItto here, please just import the value from the config file. (Also everywhere else in this PR where this repeats.)